### PR TITLE
feat(bigquery): expose CMEK functionality  for dataset

### PR DESCRIPTION
**Pull request by HemangChothani**
related to issue [9251]

### CODE REVIEW

File: bigquery/google/cloud/bigquery/dataset.py [hunk 1]
```diff
@@ -547,6 +549,30 @@ def labels(self, value):
...
+        Custom encryption configuration (e.g., Cloud KMS keys) or :data:`None`
+        if using default encryption.
+
+        See `protecting data with Cloud KMS keys
+        <https://cloud.google.com/bigquery/docs/customer-managed-encryption>`_
+        in the BigQuery documentation.
+        """
+        prop = self._properties.get("encryptionConfiguration")
+        if prop is not None:
+            prop = EncryptionConfiguration.from_api_repr(prop)
```
Comment (mf2199) [inline bigquery/google/cloud/bigquery/dataset.py]: Can we simply write
```
        if prop:
            prop = EncryptionConfiguration.from_api_repr(prop)
```
?

File: bigquery/google/cloud/bigquery/dataset.py [hunk 2]
```diff
@@ -547,6 +549,30 @@ def labels(self, value):
...
+        return prop
+
+    @encryption_configuration.setter
+    def encryption_configuration(self, value):
+        api_repr = value
+        if value is not None:
+            api_repr = value.to_api_repr()
```
Comment (mf2199) [inline bigquery/google/cloud/bigquery/dataset.py]: Same here?

File: bigquery/google/cloud/bigquery/dataset.py (hunk 1 above)
Comment (HemangChothani) [reply bigquery/google/cloud/bigquery/dataset.py]: Yes we can, but same properties already written in table class so i didn't change it .

File: bigquery/google/cloud/bigquery/dataset.py (hunk 2 above)
Comment (HemangChothani) [reply bigquery/google/cloud/bigquery/dataset.py]: same as.

**Comment by HemangChothani** [comment]
@Emar-Kar  in docs there is "defaultEncryptionConfiguration" Parameter  that name i need to change but it takes same object of class EncryptionConfiguration.

**Comment by Emar-Kar** [comment]
Yeah, that's what I mean. Nevertheless, it is not the same thing as `EncryptionConfiguration` in the **table** or the **model** or the **job**. Thus all your changes should set `defaultEncryptionConfiguration`. It will set the encryption key for all tables in the dataset but not for the dataset itself. I think you misunderstand the function of this parameter.

File: bigquery/google/cloud/bigquery/dataset.py [hunk 3]
```diff
@@ -552,7 +552,7 @@ def labels(self, value):
     @property
     def encryption_configuration(self):
         """google.cloud.bigquery.table.EncryptionConfiguration: Custom
-        encryption configuration for the table.
+        encryption configuration for the dataset.
```
Comment (Emar-Kar) [inline bigquery/google/cloud/bigquery/dataset.py]: This change has no sense. It is not the dataset encryption configuration.

File: bigquery/google/cloud/bigquery/dataset.py [hunk 4]
```diff
@@ -363,6 +364,7 @@ class Dataset(object):
         "created": "creationTime",
         "default_table_expiration_ms": "defaultTableExpirationMs",
         "friendly_name": "friendlyName",
+        "encryption_configuration": "defaultEncryptionConfiguration",
```
Comment (Emar-Kar) [inline bigquery/google/cloud/bigquery/dataset.py]: This becomes really confusing. I think it should be called `default_encryption_configuration`.

**Pull request closed without merging**